app/components/autocomplete.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from "react";
+import {
+  useCombobox,
+  UseComboboxState,
+  UseComboboxStateChangeOptions,
+  UseComboboxStateChange,
+} from "downshift";
+import { useState } from "react";
+import { Box, BoxProps, Flex, FlexProps, Input } from "@theme-ui/components";
+import { Themed } from "@theme-ui/mdx";
+
+const menuStyles = {
+  listStyleType: "none",
+  marginLeft: 0,
+};
+
+function AutocompleteList({ ...boxProps }: BoxProps) {
+  return (
+    <Box
+      as="ul"
+      {...boxProps}
+      sx={{
+        pl: 0,
+        listStyleType: "none",
+        ...boxProps.sx,
+        bg: "monochrome100",
+        boxShadow: "primary",
+        borderRadius: 10,
+        overflow: "hidden",
+      }}
+    >
+      {boxProps.children}
+    </Box>
+  );
+}
+
+function AutocompleteResult({
+  icon,
+  ...boxProps
+}: { icon: React.ReactNode } & BoxProps) {
+  return (
+    <Flex
+      as="li"
+      {...boxProps}
+      sx={{
+        alignItems: "center",
+        px: 4,
+        py: 4,
+      }}
+    >
+      {icon ? <Box sx={{ width: 24, height: 24 }}>{icon}</Box> : null}
```

Comment:
nit: icon && <Box...></Box>?

Reply:
Ah yes, I always do that, not sure why, I agree.

Reply:
I will keep it this way as it is not so different and that I prefer to be able to quickly change the null to something :)